core/src/test/java/org/apache/iceberg/rest/responses/TestListTablesResponse.java

```diff
@@ -105,6 +105,23 @@ public void testBuilderDoesNotCreateInvalidObjects() {
         .hasMessage("Invalid table identifier: null");
   }
 
+  @Test
+  public void testWithNullPaginationToken() {
+    ListTablesResponse response =
+        ListTablesResponse.builder().addAll(IDENTIFIERS).nextPageToken(null).build();
+    Assertions.assertThat(response.nextPageToken()).isNull();
+    Assertions.assertThat(response.identifiers()).isEqualTo(IDENTIFIERS);
+  }
+
+  @Test
+  public void testWithPaginationToken() {
```

Comment:
this should also do a JSON round-trip check similar to `testRoundTripSerDe()`

Reply:
will add this